src/Directory.Build.props

```diff
@@ -1,29 +1,47 @@
 <Project>
-  <PropertyGroup>
+
+  <PropertyGroup Label="Build configuration">
+    <LangVersion>7.3</LangVersion>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)..\obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
     <OutputPath>$(MSBuildThisFileDirectory)..\bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
     <PackageOutputPath>$(MSBuildThisFileDirectory)..\bin\Packages\$(Configuration)\</PackageOutputPath>
+    <IsTestProject>$(MSBuildProjectName.Contains('Test'))</IsTestProject>
+    <IsPackable Condition=" $(IsTestProject) ">false</IsPackable>
+    <GenerateDocumentationFile Condition=" !$(IsTestProject) ">true</GenerateDocumentationFile>
```

Comment:
We should *always* generate a documentation file. Otherwise C# doesn't run its analysis on the xml doc comments that *do* exist and flag syntax and reference errors. 
It's fine though to use `<NoWarn>$(NoWarn);CS1591</NoWarn>` in test projects so that no documentation is *required* in test projects though. 
I believe there is even a StyleCop rule that calls out this pattern. Is it disabled?

Reply:
I've applied the suggestion.